pymeasure/display/widgets/image_widget.py

```diff
@@ -35,44 +35,67 @@
 
 
 class ImageWidget(TabWidget, QtWidgets.QWidget):
-    """ Extends the :class:`ImageFrame<pymeasure.display.widgets.image_frame.ImageFrame>`
+    """Extends the :class:`ImageFrame<pymeasure.display.widgets.image_frame.ImageFrame>`
     to allow different columns of the data to be dynamically chosen
     """
 
-    def __init__(self, name, columns, x_axis, y_axis, z_axis=None, refresh_time=0.2,
-                 check_status=True, parent=None):
+    def __init__(
+        self,
+        name,
+        columns,
+        x_axis,
+        y_axis,
+        z_axis=None,
+        refresh_time=0.2,
+        check_status=True,
+        colormap="viridis",
+        force_reload=False,
+        parent=None,
+    ):
         super().__init__(name, parent)
         self.columns = columns
         self.refresh_time = refresh_time
         self.check_status = check_status
         self.x_axis = x_axis
         self.y_axis = y_axis
+        self.colormap = colormap
         self._setup_ui()
         self._layout()
         if z_axis is not None:
             self.columns_z.setCurrentIndex(self.columns_z.findText(z_axis))
             self.image_frame.change_z_axis(z_axis)
+        self.force_reload = force_reload
 
     def _setup_ui(self):
         self.columns_z_label = QtWidgets.QLabel(self)
-        self.columns_z_label.setMaximumSize(QtCore.QSize(45, 16777215))
-        self.columns_z_label.setText('Z Axis:')
+        self.columns_z_label.setText("Z Axis:")
 
         self.columns_z = QtWidgets.QComboBox(self)
         for column in self.columns:
             self.columns_z.addItem(column)
-        self.columns_z.activated.connect(self.update_z_column)
+        self.columns_z.currentIndexChanged.connect(self.update_z_column)
+
+        list_of_maps = pg.colormap.listMaps()
+        list_of_maps = sorted(list_of_maps, key=lambda x: x.swapcase())
+        self.colormaps_label = QtWidgets.QLabel(self)
+        self.colormaps_label.setText("Colormap:")
+
+        self.colormaps = QtWidgets.QComboBox(self)
+        self.colormaps.addItems(list_of_maps)
 
         self.image_frame = ImageFrame(
-            self.x_axis,
-            self.y_axis,
-            self.columns[0],
-            self.refresh_time,
-            self.check_status
+            self.x_axis, self.y_axis, self.columns[0], self.refresh_time, self.check_status
         )
-        self.updated = self.image_frame.updated
         self.plot = self.image_frame.plot
-        self.columns_z.setCurrentIndex(2)
+        self.colorbar = pg.ColorBarItem(
+            colorMap=self.colormap,
+            interactive=False,
```

Comment:
I wanted to say "why would you disable this, seems quite a nice feature", until I tried it, but it is extremely finicky..... not really usefull it seems... 😞